linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/GenIncOptimizer.scala

```diff
@@ -66,8 +66,9 @@ abstract class GenIncOptimizer private[optimizer] (config: CommonPhaseConfig) {
 
   private var objectClass: Class = _
   private val classes = CollOps.emptyMap[String, Class]
-  private val statics = CollOps.emptyParMap[String, StaticsNamespace]
-  private val defaults = CollOps.emptyParMap[String, Defaults]
+
+  private val staticLikes =
+    CollOps.emptyParMap[String, Array[StaticLikeNamespace]]
```

Comment:
Would it be possible to type this as `Array[MethodContainer]` instead and get rid of the strange empty `StaticLikeNamespaces` for classes (and put the `Class` instead)?

Reply:
Unfortunately, if I do that, it becomes very difficult to reason about what happens when a given `encodedName` transitions from being a class to being something else in an incremental run, or vice versa. The timing of when to delete/change/add things becomes very complicated.